geode-apis-compatible-with-redis/src/distributedTest/java/org/apache/geode/redis/internal/executor/sortedset/ZRemDUnitTest.java

```diff
@@ -0,0 +1,266 @@
+    }
+  }
+
+  private void doZRem1(AtomicInteger total) {
+    for (int i = 0; i < setSize; i++) {
+      long count = jedis.zrem(sortedSetKey, baseName + i);
+      total.addAndGet((int) count);
+    }
+  }
+
+  @Test
+  public void zRemRemovesMembersFromSortedSetAfterPrimaryShutsDown() {
+    Map<String, Double> memberScoreMap = makeMemberScoreMap(setSize);
+    jedis.zadd(sortedSetKey, memberScoreMap);
+    verifyDataExist(memberScoreMap);
+
+    stopNodeWithPrimaryBucketOfTheKey(false);
+
+    doZRemWithRetries(memberScoreMap);
+
+    verifyDataNotExist(memberScoreMap);
+    assertThat(jedis.exists(sortedSetKey)).isFalse();
+  }
+
+  private void doZRemWithRetries(Map<String, Double> map) {
+    int maxRetryAttempts = 10;
+    int retryAttempts = 0;
+    while (!zRemWithRetries(map, retryAttempts, maxRetryAttempts)) {
+      retryAttempts++;
+    }
+  }
+
+  private boolean zRemWithRetries(Map<String, Double> map, int retries, int maxRetries) {
+    long removed;
+    try {
+      removed = jedis.zrem(sortedSetKey, map.keySet().toArray(new String[] {}));
+    } catch (JedisClusterMaxAttemptsException e) {
+      if (retries < maxRetries) {
+        return false;
+      }
+      throw e;
+    }
+    assertThat(removed).isEqualTo(map.size());
+    return true;
+  }
+
+  private void doZRem(Map<String, Double> map) {
+    long removed = jedis.zrem(sortedSetKey, map.keySet().toArray(new String[] {}));
+    assertThat(removed).isEqualTo(map.size());
+  }
+
+  @Test
+  @Ignore("Fails due to GEODE-9310")
+  public void zRemCanRemoveMembersFromSortedSetDuringPrimaryIsCrashed() throws Exception {
+    int mapSize = 300;
+    Map<String, Double> memberScoreMap = makeMemberScoreMap(mapSize);
+
+    jedis.zadd(sortedSetKey, memberScoreMap);
+    verifyDataExist(memberScoreMap);
+
+    int number = 10;
+    String memberNotRemoved = baseName + number;
+    memberScoreMap.remove(memberNotRemoved);
+
+    Future<Void> future1 = executor.submit(() -> doZRem(memberScoreMap));
+    Future<Void> future2 = executor.submit(() -> stopNodeWithPrimaryBucketOfTheKey(true));
+
+    future1.get();
+    future2.get();
+
+    GeodeAwaitility.await().until(() -> verifyDataNotExist(memberScoreMap));
+    assertThat(jedis.exists(sortedSetKey)).isTrue();
+  }
+
+  private void verifyDataExist(Map<String, Double> memberScoreMap) {
+    for (String member : memberScoreMap.keySet()) {
+      Double score = jedis.zscore(sortedSetKey, member);
+      assertThat(score).isEqualTo(memberScoreMap.get(member));
+    }
+  }
+
+  private boolean verifyDataNotExist(Map<String, Double> memberScoreMap) {
+    try {
+      for (String member : memberScoreMap.keySet()) {
+        Double score = jedis.zscore(sortedSetKey, member);
+        assertThat(score).isNull();
+      }
+    } catch (JedisClusterMaxAttemptsException e) {
+      return false;
+    }
+    return true;
+  }
+
+  private void stopNodeWithPrimaryBucketOfTheKey(boolean isCrash) {
+    int numOfServers = 4;
```

Comment:
Rather than hard-coding this value, it might be better to add the servers to a `List<MemberVM>` in the `setUp()` method and iterate through that here.